sdk/identity/Azure.Identity/tests/VisualStudioCodeCredentialLiveTests.cs

```diff
@@ -0,0 +1,411 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Core.Testing;
+using Microsoft.Identity.Client;
+using NUnit.Framework;
+
+namespace Azure.Identity.Tests
+{
+    public class VisualStudioCodeCredentialLiveTests : RecordedTestBase
+    {
+        private const string ExpectedServiceName = "VS Code Azure";
+
+        public VisualStudioCodeCredentialLiveTests(bool isAsync) : base(isAsync)
+        {
+            Matcher.ExcludeHeaders.Add("Content-Length");
```

Comment:
For curiosity sake why are we excluding content length?

Reply:
`IdentityRecordedTestSanitizer` changes request body, so computed "Content-Length" will be incorrect.

Reply:
Hm! https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/tests/TestFramework/RecordedTestSanitizer.cs#L88

Reply:
@schaabs , should we change it?

Reply:
In theory it should just work, can you debug and see why that codepaths is not getting hit?

Reply:
The identity tests have their own sanitizer, and it should update Content-Length as well.

https://github.com/Azure/azure-sdk-for-net/blob/854baa748ef63c3177c24d7ebeef2a3585332d51/sdk/identity/Azure.Identity/tests/IdentityRecordedTestSanitizer.cs#L28-L30

If we can't figure out why we still need to omit this from the matcher let's not block merging this. Just file an issue to investigate.

Reply:
Error looks like this:
```
Header differences:
    <Content-Length> values differ, request <181>, record <9>
```
So it is playback `Content-Length` that isn't sanitized.